staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -199,7 +199,7 @@ type sharedIndexInformer struct {
 	clock clock.Clock
 
 	started, stopped bool
-	startedLock      sync.Mutex
+	startedLock      sync.RWMutex
```

Comment:
Technically this change is correct. But does that really buy us something? Have you even seen contention on this lock that could be cause by those two methods?

Reply:
Let me run some tests and see if I can come up with performance numbers showing the effectiveness.

Reply:
+1 @wojtek-t comment. We know that RWMutex is more expensive than a regular mutex. So we need to be sure that we gaining enough from read contention to offset the additional locking cost.